version: '3.8'
services:
  db:
    build:
      context: .
      dockerfile: .docker/postgres/Dockerfile
    ports:
      - "5432:5432"
    environment:
      - POSTGRES_USER=${POSTGRES_USER}
      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
      - POSTGRES_DB=${POSTGRES_DB}
    networks:
      app_net:
        ipv4_address: 172.16.0.2
    healthcheck:
      test:
          - CMD-SHELL
          - pg_isready
          - -d
          - ${POSTGRES_DB}
          - -U
          - ${POSTGRES_USER}
          - -q
      interval: 15s
      timeout: 5s
      retries: 5
      start_period: 5s
  web:
    depends_on:
        db:
            condition: service_healthy
        redis:
            condition: service_started
    build:
      context: .
      dockerfile: .docker/web/Dockerfile
    ports:
      - "8080:4444"
    networks:
      app_net:
        ipv4_address: 172.16.0.3
    volumes:
        - ./src:/home/django/src
    healthcheck:
      test:
          - CMD
          - curl
          - --fail
          - http://localhost:4444
      interval: 15s
      timeout: 5s
      retries: 5
      start_period: 5s
  redis:
    image: redis:latest
    ports:
      - "6379:6379"
    networks:
      app_net:
        ipv4_address: 172.16.0.4

networks:
  app_net:
    ipam:
      driver: default
      config:
        - subnet: "172.16.0.0/24"
        - subnet: "2001:3984:3989::/64"
